common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -936,6 +941,32 @@ private AccountRecord getAccountWithFRTIfAvailable(@NonNull final SilentTokenCom
         return null;
     }
 
+    @Nullable
+    private AccountRecord getAccountFromFociCacheUsingLocalAccountId(@NonNull final BrokerOAuth2TokenCache brokerOAuth2TokenCache,
+                                                                     @NonNull final String localAccountId) {
+        final String methodName = ":getAccountFromFociCacheUsingLocalAccountId";
+        AccountRecord targetAccount = null;
+        final List<CacheRecord> fociCacheRecords = brokerOAuth2TokenCache.getFociCacheRecords();
+
+        if (fociCacheRecords.size() > 0) {
+            Logger.info(
+                    TAG + methodName,
+                    "There is (are) ["
+                            + fociCacheRecords.size()
+                            + "] Foci in the broker cache."
+            );
+
+            for (final ICacheRecord cacheRecord : fociCacheRecords) {
+                if (cacheRecord.getAccount() != null && localAccountId
+                        .equalsIgnoreCase(cacheRecord.getAccount().getLocalAccountId())) {
+                    targetAccount = cacheRecord.getAccount();
```

Comment:
should we return here? Do we need to keep going through the loop?

Reply:
Removed this function as I made changes in broker repo to override getCachedAccountRecordFromAllCaches. It is in this method, I am reusing the existing BrokerUtil.getCacheRecordListFromBrokerCache() method which internally calls mPlatformComponents.isAuthorizedToShareTokens()